File: packages/ts/src/config/networks.ts
/**
 * Network presets for the QoreChain SDK.
 *
 * Both the `testnet` and `mainnet` presets are fully populated and live. Their
 * endpoints default to localhost ports so the SDK works out of the box against a
 * locally running node; callers override these with real hostnames when creating
 * a client. {@link getNetwork} returns either preset.
 */

/** Bech32 human-readable prefixes used across QoreChain address types. */
export interface Bech32Prefixes {
  /** Prefix for account addresses (e.g. `qor1...`). */
  account: string;
  /** Prefix for validator operator addresses (e.g. `qorvaloper1...`). */
  validator: string;
  /** Prefix for validator consensus addresses (e.g. `qorvalcons1...`). */
  consensus: string;
}

/** Display and base denomination metadata for the network's staking coin. */
export interface CoinInfo {
  /** Human-facing denomination (e.g. `QOR`). */
  display: string;
  /** Base (smallest) denomination used on-chain (e.g. `uqor`). */
  base: string;
  /** Decimal exponent relating base to display (1 display = 10^exponent base). */
  exponent: number;
}

/** Service endpoints for talking to a network across its supported VMs. */
export interface NetworkEndpoints {
  /** Cosmos SDK REST (LCD) endpoint. */
  rest: string;
  /** Cosmos SDK gRPC endpoint. */
  grpc: string;
  /** Consensus RPC endpoint. */
  rpc: string;
  /** EVM JSON-RPC HTTP endpoint. */
  evmRpc: string;
  /** EVM JSON-RPC WebSocket endpoint. */
  evmWs: string;
  /** SVM JSON-RPC endpoint. */
  svmRpc: string;
}

/** A fully described network preset. */
export interface NetworkConfig {
  /** Canonical preset name. */
  name: string;
  /** Whether the network is live and usable without custom endpoints. */
  live: boolean;
  /** Chain ID. */
  chainId: string;
  /** Bech32 prefixes for address encoding. */
  bech32: Bech32Prefixes;
  /** Staking coin metadata. */
  coin: CoinInfo;
  /** Default endpoints. */
  endpoints: NetworkEndpoints;
}

/** Known network preset names. */
export type NetworkName = "testnet" | "mainnet";

/**
 * QoreChain uses the same token and address prefixes on every network, so these
 * are shared (not invented) values across the presets below.
 */
const BECH32: Bech32Prefixes = {
  account: "qor",
  validator: "qorvaloper",
  consensus: "qorvalcons",
};

const COIN: CoinInfo = {
  display: "QOR",
  base: "uqor",
  exponent: 6,
};

/** The set of built-in network presets, keyed by name. */
export const NETWORKS: Record<NetworkName, NetworkConfig> = {
  testnet: {
    name: "testnet",
    live: true,
    chainId: "qorechain-diana",
    bech32: BECH32,
    coin: COIN,
    endpoints: {
      rest: "http://localhost:1317",
      grpc: "http://localhost:9090",
      rpc: "http://localhost:26657",
      evmRpc: "http://localhost:8545",
      evmWs: "ws://localhost:8546",
      svmRpc: "http://localhost:8899",
    },
  },
  mainnet: {
    name: "mainnet",
    live: true,
    chainId: "qorechain-vladi",
    bech32: BECH32,
    coin: COIN,
    endpoints: {
      rest: "http://localhost:1317",
      grpc: "http://localhost:9090",
      rpc: "http://localhost:26657",
      evmRpc: "http://localhost:8545",
      evmWs: "ws://localhost:8546",
      svmRpc: "http://localhost:8899",
    },
  },
};

/**
 * Resolve a network preset by name.
 *
 * @returns The requested {@link NetworkConfig}, guaranteed to be live and usable.
 */
export function getNetwork(name: NetworkName): NetworkConfig {
  return NETWORKS[name];
}

/** List the known network preset names without triggering any liveness check. */
export function listNetworks(): NetworkName[] {
  return Object.keys(NETWORKS) as NetworkName[];
}

File: packages/java/src/main/java/io/github/qorechain/networks/Networks.java
package io.github.qorechain.networks;

import java.util.Collections;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

/**
 * Built-in QoreChain network presets.
 *
 * <p>{@code testnet} is {@code qorechain-diana} and {@code mainnet} is
 * {@code qorechain-vladi}; both are live and share the {@code qor}/{@code uqor}
 * coin and bech32 prefixes. Endpoints default to localhost ports.
 */
public final class Networks {

    private Networks() {}

    /**
     * QoreChain uses the same token and address prefixes on every network, so
     * these are shared values across the presets.
     */
    private static final NetworkConfig.Bech32Prefixes BECH32 =
            new NetworkConfig.Bech32Prefixes("qor", "qorvaloper", "qorvalcons");

    private static final NetworkConfig.CoinInfo COIN =
            new NetworkConfig.CoinInfo("QOR", "uqor", 6);

    private static NetworkConfig.NetworkEndpoints localhostEndpoints() {
        return new NetworkConfig.NetworkEndpoints(
                "http://localhost:1317",
                "http://localhost:9090",
                "http://localhost:26657",
                "http://localhost:8545",
                "ws://localhost:8546",
                "http://localhost:8899");
    }

    private static final Map<String, NetworkConfig> NETWORKS;

    static {
        Map<String, NetworkConfig> m = new LinkedHashMap<>();
        m.put(
                "testnet",
                new NetworkConfig(
                        "testnet", true, "qorechain-diana", BECH32, COIN, localhostEndpoints()));
        m.put(
                "mainnet",
                new NetworkConfig(
                        "mainnet", true, "qorechain-vladi", BECH32, COIN, localhostEndpoints()));
        NETWORKS = Collections.unmodifiableMap(m);
    }

    /**
     * Resolve a network preset by name ({@code "testnet"} or {@code "mainnet"}).
     *
     * @throws IllegalArgumentException if the name is unknown.
     */
    public static NetworkConfig get(String name) {
        NetworkConfig cfg = NETWORKS.get(name);
        if (cfg == null) {
            throw new IllegalArgumentException("unknown network: " + name);
        }
        return cfg;
    }

    /** List the known network preset names. */
    public static List<String> list() {
        return List.copyOf(NETWORKS.keySet());
    }
}
